src/appleseed/renderer/modeling/light/sunlight.cpp

```diff
@@ -125,6 +127,23 @@ namespace
             // Evaluate uniform inputs.
             m_inputs.evaluate_uniforms(&m_values);
 
+
+            Source* distance_src = get_inputs().source("distance");
+            assert(distance_src != nullptr);
+            if (distance_src->is_uniform())
+                distance_src->evaluate_uniform(m_values.m_distance);
+            else
+            {
+                RENDERER_LOG_WARNING(
+                    "Distance between Sun and scene \"%s\" is not uniform.",
+                    get_path().c_str());
+            }
+
+            // Compute SunSolidAngle that depends on distance between Sun and scene.
+            // angular_diameter = 2 * arctan(sun_radius / (distance))
+            SunSolidAngle = 2 * M_PI * (1 - cos(atan(SunRadius / m_values.m_distance)));
```

Comment:
The problem of `M_PI` is that it's a `double` while we need a `float`. This will fail to compile with Visual Studio (among others). Instead you can use `TwoPi<float>()` in place of `2 * M_PI`.

Reply:
Are you sure about the formula here? The argument of `cos()` should be the radius of the sun, but you passed it the angular radius (which isn't the same thing).

Also, if I plug the default values into your formula (`SunRadius` = 0.6957, `m_values.m_distance` = 149.6) I get `SunSolidAngle` = 6.79e-5 instead of 6.87e-5.

Reply:
`angular diameter` = 2 * `atan(diameter / (2 * distance))`
`angular diameter` = 2 * atan(radius / distance)` = 2 * `theta` (in wikipedia)
`sun solid angle` = 2 * PI * (1-cos(theta))

this is how i make formula there. i think that difference came because all value(`SunRadius`, `distance`, and so on) are approximate value, not real..

Reply:
@dictoon 
Value 6.87e-5 assumes that `angular diameter` is 0.00935 (not 0.009301), 
However `angular diameter` in [wikipedia](https://en.wikipedia.org/wiki/Angular_diameter) is 31′31″ – 32′33″ (is similar with 0.00916, 0.00946)